### PR TITLE
feat(ai): add slide boundaries and speaker notes to PowerPoint extraction

### LINKED ISSUE #127: feat(files): Add PowerPoint (PPT/PPTX) file processing
## Context

The file processing pipeline currently handles PDF, DOC/DOCX, TXT, Markdown, JSON, and CSV. **PowerPoint files (PPT/PPTX) are not supported**, which is a gap for professors who frequently upload lecture slides as their primary teaching material.

PPTX files contain rich content: slide text, speaker notes, table data, and embedded images. All of this content should be extractable and queryable by students.

## Relevant Code

| File | Purpose |
|------|---------|
| `apps/web/src/server/routers/files/validation.ts` | File type validation — add PPTX/PPT MIME types here |
| `packages/ai/src/rag-service.ts` | Text extraction functions — add `extractTextFromPowerPoint` here |
| `apps/web/src/lib/file-processor.ts` | Processing pipeline — wire up new extractor via MIME type switch |

### Existing pattern for Word documents (similar structure to follow):

The Word extraction uses `mammoth` to extract raw text:

```typescript
export async function extractTextFromWord(buffer: Buffer): Promise<string> {
  const result = await mammoth.extractRawText({ buffer });
  const text = result.value;
  // ... sanitize and return
}
```

Follow this same pattern: accept buffer, extract text, sanitize, return.

### 1. Add PPTX/PPT file type support
- Add MIME types to `ALLOWED_FILE_TYPES`:
  - `application/vnd.openxmlformats-officedocument.presentationml.presentation` (`.pptx`)
  - `application/vnd.ms-powerpoint` (`.ppt`)
- Add extension-to-MIME mappings for `.pptx` and `.ppt`

### 2. Implement text extraction
- Create `extractTextFromPowerPoint(buffer: Buffer): Promise<string>` in `rag-service.ts`
- Extract **all text content** from slides:
  - Slide body text
  - Speaker notes
  - Table cell text
  - Text in grouped shapes
- Structure extracted text with clear slide boundaries (e.g., `--- Slide 1 ---\n...`)
- Sanitize output using the existing `sanitizeText` helper

### 3. Wire into the processing pipeline
- Add PPTX/PPT to the MIME type switch in `file-processor.ts`
- Ensure progress tracking works correctly

## Acceptance Criteria

- [ ] PPTX and PPT files can be uploaded and pass validation
- [ ] Text is extracted from slide body, speaker notes, and tables
- [ ] Extracted text preserves slide structure (slide numbers/boundaries)
- [ ] Empty presentations throw a descriptive error
- [ ] Extracted text flows through chunking → embedding → storage correctly
- [ ] Unit tests cover: normal PPTX, PPTX with only speaker notes, empty PPTX, corrupted file
- [ ] `npm run check-types` and `npm run lint` pass

## Suggested Libraries (Starting Points)

> These are starting points for research — feel free to choose alternatives if they're a better fit. Please briefly explain your library choices in the PR description.

- **[`officeparser`](https://github.com/harshankur/officeParser)** (v6.0.4, ~215K weekly downloads, Jan 2026 release) — All-in-one parser for DOCX, PPTX, XLSX, PDF, RTF. Produces a structured AST with metadata and formatting. TypeScript support. Actively maintained.
- **[`node-pptx-parser`](https://github.com/Mirza-Glitch/node-pptx-parser)** (v1.0.1, ~7.6K weekly downloads) — Focused PPTX parser with text extraction and formatting preservation. TypeScript. Lighter weight.
- **DIY with `jszip`** — PPTX is a ZIP of XML files. Slide content is in `ppt/slides/slide*.xml` and notes in `ppt/notesSlides/notesSlide*.xml`. You can parse these directly if you prefer minimal dependencies.

### Important notes
- **Legacy PPT**: `.ppt` (OLE2 format) is significantly harder to parse than `.pptx`. If no good library handles it, it's acceptable to only support `.pptx` and document the limitation.
- **Slide ordering**: Extract slides in numerical order (slide1.xml, slide2.xml, ...).
- **Chunk boundaries**: The `--- Slide N ---` markers help the chunker split at semantic boundaries since `\n\n` is the first separator tried.

## Contributing Guidelines

- Follow the coding standards in `AGENTS.md` (strict TypeScript, no `any`, kebab-case files, etc.)
- PR title must use conventional commit format: `feat(files): add PowerPoint file processing`
- Include tests for the new extraction function
- Run `npm run lint && npm run check-types` before submitting
- Link this issue in your PR description

### PR DESCRIPTION
**Pull request by AnuSingh18**
https://github.com/akhileshrangani4/teachanything/issues/127

 ## Summary
  - Rewrite `extractPowerPoint()` to traverse the officeparser AST instead of `toText()`, preserving slide boundaries and speaker notes in extracted text
  - Output uses `--- Slide N ---` markers and `[Speaker Notes]` labels for structured chunking
  - Add Vitest test framework to `packages/ai` with 6 unit tests
  - Add `.idea/` and `.claude/settings.local.json` to `.gitignore`

Closes #127

  ## Test plan
  - [x] All 6 Vitest tests pass
  - [x] `npm run check-types` passes
  - [x] `npm run lint` passes
  - [x] Manual test: uploaded .pptx file, verified slide markers in processed chunks

### CODE REVIEW

**Review by akhileshrangani4 — CHANGES_REQUESTED** [review]
Solid feature addition — the AST traversal approach for PowerPoint extraction is much better than `toText()`, and the tests cover the important edge cases well. A few issues to address before merging.

**Review by akhileshrangani4 — COMMENTED** [review]
A couple more minor notes:

File: apps/web/src/components/dashboard/files/file-constants.ts [hunk 1]
```diff
@@ -11,6 +11,8 @@ export const ALLOWED_FILE_TYPES = [
   "application/pdf",
   "application/vnd.openxmlformats-officedocument.wordprocessingml.document",
   "application/msword",
+  "application/vnd.openxmlformats-officedocument.presentationml.presentation",
+  "application/vnd.ms-powerpoint",
```
[the remark this answers is not in the document]
Comment (AnuSingh18) [reply apps/web/src/components/dashboard/files/file-constants.ts]: changes done

**Review by akhileshrangani4 — APPROVED** [review]

**Comment by AnuSingh18** [comment]
@akhileshrangani4  Thanks for fixes.
